backend/src/modules/lobby/lobbyRoutes.js

```diff
@@ -59,15 +74,47 @@ router.put(
         error: formatErrors(errors),
       });
     }
+    const [lobbyId, invitationId] = Hashids.decode(lobbyHash);
+    const invitation = await dbConnection.query(
+      'SELECT invitation_id, email FROM invitation WHERE invitation_id = ? AND lobby_id = ?;',
+      [invitationId, lobbyId]
+    );
+    if (!invitation || invitation.length < 1) {
+      return res.status(422).json({
+        error: '422: Invitation expired or token is not valid.',
+      });
+    }
+    const userDetail = await dbConnection.query(
+      'SELECT user_id, email, nickname FROM users WHERE email = ? AND active = true LIMIT 1;',
+      [invitation[0].email]
+    );
 
-    // TODO check lobbyHash from DB and if user is already registered
-    // when user is not registered - return only { userIsRegistered: false }
-    // when user is registered - need to generate JWT token and SELECT user info and return:
-    // { userIsRegistered: true,
-    //   token: getJwtToken({ userId: 666 }),
-    //   user: { nickname: "Mocked Nickname" }
-    // }
-    res.json({ userIsRegistered: false });
+    const userIsRegistered = (userDetail.length > 0) ? true : false;
+
+    if (userIsRegistered === true) {
+      const { user_id: userId, nickname: nickName } = userDetail[0];
+      await dbConnection.query(
+        'INSERT INTO lobby_user (user_id, lobby_id) VALUES (?, ?);',
+        [userId, lobbyId]
+      );
+      await dbConnection.query(
+        'DELETE FROM invitation WHERE invitation_id = ?;',
+        [invitationId]
+      );
+      res.json({ 
+        userIsRegistered: true,
+        ...loginSuccessPayload(userId)
```

Comment:
Tohle nefunguje - mel jsme to v commentu jako naznaceni, co se ma vracet na FE, ale tahle funkce je undefined.. Je potreba token vygenerovat.

Reply:
takhle?
const token = getJwtToken({ userId });
      res.json({ 
        userIsRegistered: true,
        token: token,
        nickname: nickName,
      });